pkg/util/labels.go

```diff
@@ -7,43 +7,61 @@ import (
 	"github.com/openshift/source-to-image/pkg/api"
 )
 
-// GenerateOutputImageLabels generate the labels based on the source repository
-// informations.
+// GenerateOutputImageLabels generate the labels based on the s2i Config
+// and source repository informations.
 func GenerateOutputImageLabels(info *api.SourceInfo, config *api.Config) map[string]string {
-	result := map[string]string{}
+	labels := map[string]string{}
+	var buildNamespace string
+	if len(config.BuildNamespaceLabel) > 0 {
+		buildNamespace = config.BuildNamespaceLabel
+	} else {
+		buildNamespace = api.DefaultNamespace
+	}
+
+	labels = GenerateLabelsFromConfig(labels, config, buildNamespace)
+	labels = GenerateLabelsFromSourceInfo(labels, info, buildNamespace)
+	return labels
+}
 
+// GenerateLabelsFromConfig generate the labels based on build s2i Config
+func GenerateLabelsFromConfig(labels map[string]string, config *api.Config, buildNamespace string) map[string]string {
 	if len(config.Description) > 0 {
-		result[api.KubernetesNamespace+"description"] = config.Description
+		labels[api.KubernetesNamespace+"description"] = config.Description
```

Comment:
You're not using the `buildNamespace` parameter, you're passing here. Besides shorten it to just `namespace`.